metadata-models/src/main/pegasus/com/linkedin/common/BaseFieldMapping.pdl

```diff
@@ -0,0 +1,21 @@
+namespace com.linkedin.common
+
+/**
+ * Base model representing field mappings
+ */
+record BaseFieldMapping {
+  /**
+   * Audit stamp containing who reported the field mapping and when
+   */
+  created: optional AuditStamp
+
+  /**
+   * Type of the transformation involved in generating destination field from source fields. e.g., BLACKBOX, IDENTITY, CUSOTM
+   */
+  tranformationType: optional FieldTransformationType = "BLACKBOX"
```

Comment:
Another option here is to create a separate model `Transformation` and encapsulate both the type and additional info into that model so you can evolve that model without needing to update BaseFieldMapping with too fields.

Reply:
Yeah ... I agree this is a good idea. We can different models for Identity, Blackbox and UDF . Each of them can grow independent of each other.